File: backend/src/main/java/com/syit/cpc/entity/Notification.java
package com.syit.cpc.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 站内信通知实体类
 * 对应数据库表: notifications
 */
@Data
@TableName("notifications")
public class Notification {

    @TableId(type = IdType.AUTO)
    private Long id;

    /**
     * 接收用户ID
     */
    private Long userId;

    /**
     * 通知类型: promotion_approved, promotion_rejected
     */
    private String type;

    /**
     * 通知标题
     */
    private String title;

    /**
     * 通知内容
     */
    private String content;

    /**
     * 是否已读: 0-未读, 1-已读
     */
    private Integer isRead;

    /**
     * 关联业务ID（如转正申请ID）
     */
    private Long relatedId;

    /**
     * 创建时间
     */
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    /**
     * 阅读时间
     */
    private LocalDateTime readAt;

    /**
     * 逻辑删除: 0-未删除, 1-已删除
     */
    @TableLogic
    private Integer deleted;
}

File: frontend/src/types/notification.ts
/**
 * 通知类型枚举
 */
export enum NotificationType {
  /** 转正通过 */
  PROMOTION_APPROVED = 'promotion_approved',
  /** 转正拒绝 */
  PROMOTION_REJECTED = 'promotion_rejected',
}

/**
 * 通知项接口
 */
export interface NotificationItem {
  id: number
  type: string
  title: string
  content: string
  isRead: number
  relatedId: number | null
  createdAt: string
  readAt: string | null
}

/**
 * 通知列表响应
 */
export interface NotificationListResponse {
  total: number
  pages: number
  current: number
  size: number
  records: NotificationItem[]
}
